tests/integration/nm/vxlan_test.py

```diff
@@ -0,0 +1,128 @@
+#
+# Copyright (c) 2018-2019 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+from contextlib import contextmanager
+
+import pytest
+
+from libnmstate import nm
+from libnmstate.schema import Interface
+from libnmstate.schema import VXLAN
+
+from .testlib import mainloop_run
+
+
+def _vxlan_ifname(state):
+    return (
+        state[VXLAN.CONFIG_SUBTREE][VXLAN.BASE_IFACE]
+        + '.'
+        + str(state[VXLAN.CONFIG_SUBTREE][VXLAN.ID])
+    )
+
+
+def _get_vxlan_device(ifname):
+    nm.nmclient.client(refresh=True)
+    return nm.device.get_device_by_name(ifname)
+
+
+def _get_vxlan_current_state(ifname):
+    nmdev = _get_vxlan_device(ifname)
+    return nm.vxlan.get_info(nmdev) if nmdev else {}
+
+
+def test_create_and_remove_vxlan(eth1_up):
+    ifname = eth1_up[Interface.KEY][0][Interface.NAME]
+    vxlan_desired_state = {
+        VXLAN.CONFIG_SUBTREE: {
+            VXLAN.ID: 101,
+            VXLAN.BASE_IFACE: ifname,
+            VXLAN.REMOTE: '192.168.1.18',
+            VXLAN.DESTINATION_PORT: 4789,
+        }
+    }
+
+    with _vxlan_interface(vxlan_desired_state):
+        vxlan_ifname = _vxlan_ifname(vxlan_desired_state)
+        vxlan_current_state = _get_vxlan_current_state(vxlan_ifname)
```

Comment:
How about returning this from `_vxlan_interface`?
`with _vxlan_interface(...) as vxlan_current_state:` fits nicely.

Reply:
I am going to return just the vxlan_ifname, it does not fit so well at test_read_destionation_port_from_libnm

Reply:
I am just generating the vxlan name inside the new Vxlan class so it get encapsulated there.